autoPyTorch/pipeline/components/training/trainer/base_trainer.py

```diff
@@ -161,9 +163,15 @@ def repr_last_epoch(self) -> str:
 
 class BaseTrainerComponent(autoPyTorchTrainingComponent):
 
-    def __init__(self, random_state: Optional[Union[np.random.RandomState, int]] = None) -> None:
+    def __init__(self, random_state: Optional[np.random.RandomState] = None) -> None:
         super().__init__()
-        self.random_state = random_state
+        if random_state is None:
```

Comment:
same here? Why are we not passing the random state to the super class?

Reply:
Sorry, I missed this as the base trainer relies on the torch manual seed. 

Anyhow, it makes sense to pass it also to the constructor and I have pushed a fix for this. Thanks for the catch.